inspirehep/modules/citations/tasks.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+from invenio_celery import celery
+
+
+@celery.task
+def update_citation_count_for_records(recids):
+
+    for recid in recids:
+        update_citation_count.delay(recid)
+
+
+@celery.task
+def update_citation_count(recid):
+    from invenio_ext.es import es
+    from .utils import calculate_citation_count
+    from invenio_search.api import Query
+    query_result = Query('control_number:' + str(recid)).search().records()
```

Comment:
I suspect it's faster to just use the `_id` for retrieving a given record (which has the same id).